Detectors/TRD/simulation/include/TRDSimulation/Detector.h

```diff
@@ -87,7 +80,7 @@ class Detector : public o2::Base::DetImpl<Detector>
 
   // addHit
   template <typename T>
-  void addHit(T x, T y, T z, T time, T energy, int trackId, int detId);
+  void addHit(T x, T y, T z, T tof, int charge, int trackId, int detId);
```

Comment:
It seems like overkill to template x,y,z positions. And couldn't this cause problems if at some point one would change from float to double and tried to read old data.
Or is there a good reason for this?

Reply:
I think this is ok until now but we should keep an eye on this. We have tried to keep it short in memory usage (I think) and used float everywhere it is possible. But for a final answer I need to review the full code to end up with the right type. Many methods that are used from TVirtualMC read only doubles or return only doubles.